test/integration/testpaths/testpath_netmask.py

```diff
@@ -0,0 +1,200 @@
+
+        cls.template = get_template(
+            cls.apiclient,
+            cls.zone.id,
+            cls.testdata["ostype"])
+
+        cls._cleanup = []
+
+        cls.skiptest = False
+
+        if cls.hypervisor.lower() not in ["xenserver"]:
+            cls.skiptest = True
+            return
+
+        try:
+
+            # Create an account
+            cls.account = Account.create(
+                cls.apiclient,
+                cls.testdata["account"],
+                domainid=cls.domain.id
+            )
+
+            # Create user api client of the account
+            cls.userapiclient = testClient.getUserApiClient(
+                UserName=cls.account.name,
+                DomainName=cls.account.domain
+            )
+
+            # Create Service offering
+            cls.service_offering = ServiceOffering.create(
+                cls.apiclient,
+                cls.testdata["service_offering"],
+            )
+
+            cls.testdata["shared_network_offering"]["specifyVlan"] = 'True'
+            cls.testdata["shared_network_offering"]["specifyIpRanges"] = 'True'
+
+            cls.shared_network_offering = NetworkOffering.create(
+                cls.apiclient,
+                cls.testdata["shared_network_offering"]
+            )
+
+            NetworkOffering.update(
+                cls.shared_network_offering,
+                cls.apiclient,
+                id=cls.shared_network_offering.id,
+                state="enabled"
+            )
+
+            cls.network = Network.create(
+                cls.apiclient,
+                cls.testdata["network2"],
+                networkofferingid=cls.shared_network_offering.id,
+                zoneid=cls.zone.id,
+                accountid=cls.account.name,
+                domainid=cls.account.domainid
+            )
+
+            cls.vm = VirtualMachine.create(
+                cls.apiclient,
+                cls.testdata["small"],
+                templateid=cls.template.id,
+                accountid=cls.account.name,
+                domainid=cls.account.domainid,
+                serviceofferingid=cls.service_offering.id,
+                zoneid=cls.zone.id,
+                networkids=cls.network.id,
+            )
+
+            cls._cleanup.extend([cls.account,
+                                 cls.service_offering,
+                                 cls.shared_network_offering])
+
+        except Exception as e:
+            cls.tearDownClass()
+            raise e
+        return
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            cleanup_resources(cls.apiclient, cls._cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+
+    def setUp(self):
+        if self.skiptest:
+            self.skipTest(
+                "Test not to be run on %s" %
+                self.hypervisor)
+        self.apiclient = self.testClient.getApiClient()
```

Comment:
Isn't this redundant (self.apiclient ) ? You should already have acess to apiclient as it is created in setupClass I guess ?